src/authgss_prot.c

```diff
@@ -101,6 +101,7 @@ xdr_rpc_gss_buf(XDR *xdrs, gss_buffer_t buf, u_int maxsize)
 	case XDR_DECODE:
 		return (xdr_rpc_gss_decode(xdrs, buf));
 	case XDR_FREE:
+		mem_free(buf->value, buf->length);
```

Comment:
I don't think this is right.  This buffer is part of the gss context (at least, it is elsewhere in the code), and it needs to be released as part of that context.

Reply:
But the decode calls xdr_bytes_decode, which does allocate memory if it isn't passed a buffer. I think the caller is obligated to not call xdr_free if it pre-allocated buffers for the decode, otherwise the XDR_FREE needs to release any memory allocated by XDR_ENCODE.

Reply:
@dang This leak is reported by leak sanitizer (This is on 1.8.0, Line numbers may not match)
Direct leak of 31312 byte(s) in 1957 object(s) allocated from:
    #0 0x7f866e7308a0 in malloc ??:?
    #1 0x7f866e255883 in gsh_malloc__ /usr/src/debug/nfs-ganesha-2.8.3/include/abstract_mem.h:78
    #2 0x7f866df87fc4 in xdr_bytes_decode /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/ntirpc/rpc/xdr_inline.h:606
    #3 0x7f866df88252 in xdr_rpc_gss_decode /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/authgss_prot.c:79
    #4 0x7f866df88322 in xdr_rpc_gss_buf /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/authgss_prot.c:100
    #5 0x7f866df883f1 in xdr_rpc_gss_cred /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/authgss_prot.c:116 (discriminator 3)
    #6 0x7f866df8a21a in _svcauth_gss /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_auth_gss.c:471
    #7 0x7f866df7403f in svc_auth_authenticate /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_auth.c:98
    #8 0x7f866e24f220 in nfs_rpc_process_request /usr/src/debug/nfs-ganesha-2.8.3/MainNFSD/nfs_worker_thread.c:1329
    #9 0x7f866e251aae in nfs_rpc_valid_NFS /usr/src/debug/nfs-ganesha-2.8.3/MainNFSD/nfs_worker_thread.c:2113
    #10 0x7f866df7ed4b in svc_vc_decode /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_vc.c:1032
    #11 0x7f866df7a806 in svc_request /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_rqst.c:793
    #12 0x7f866df7ec58 in svc_vc_recv /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_vc.c:1005
    #13 0x7f866df7a786 in svc_rqst_xprt_task /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_rqst.c:774
    #14 0x7f866df7b09c in svc_rqst_epoll_loop /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/svc_rqst.c:1089
    #15 0x7f866df84a3f in work_pool_thread /usr/src/debug/nfs-ganesha-2.8.3/libntirpc/src/work_pool.c:183
    #16 0x7f866c2e2ea4 in start_thread pthread_create.c:?
    #17 0x7f866b9e18dc in __clone ??:?